django_project/ford3/views/qualification.py

```diff
@@ -0,0 +1,20 @@
+from django.shortcuts import redirect, reverse
+from ford3.models.qualification import Qualification
+
+
+def publish_qualification(request, provider_id, qualification_id, campus_id):
+    qualification = Qualification.objects.get(pk=qualification_id)
+    qualification.published = True
+    qualification.save()
+    return redirect(reverse('show-campus',
+                            args=[str(provider_id),
+                                  str(campus_id)]))
+
+
+def unpublish_qualification(request, provider_id, qualification_id, campus_id):
```

Comment:
instead of having two view methods, you could have only one:
```
def toggle_publication(self):
  ..
  self.qualification.published = !self.qualification.published
```